docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
+`configuration options`_ that can be set.
+
+The `docs/_static/sphinx_rtd_overrides.css`_ file contains `style
+overrides`_ for the `Read the Docs Sphinx Theme`_ to customize the look
+and feel of the online documentation.
+
+Sphinx extensions
+~~~~~~~~~~~~~~~~~
+
+`PlasmaPy's documentation`_ is built with the following Sphinx_
+extensions:
+
+* `sphinx.ext.autodoc` for including documentation from docstrings
+* `sphinx.ext.intersphinx` for linking to other projects' documentation
+* `sphinx.ext.graphviz` to allow Graphviz_ graphs to be included
+* `sphinx.ext.mathjax` for math rendering with MathJax_
+* `sphinx.ext.napoleon` for allowing NumPy style docstrings
+* `sphinx.ext.todo` to support ``todo`` directives
+* nbsphinx_ for including Jupyter_ notebooks
+* `sphinx_copybutton`_ to add a "copy" button for code blocks
+* `sphinx_gallery.load_style`_ for using sphinx-gallery styles
+* `IPython.sphinxext.ipython_console_highlighting`_
+* `sphinx_changelog`_ for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy and
+  affiliated packages. Note that `plasmapy_sphinx` is expected to be
+  broken out into its own package in the future.
+
+These extensions are specified in :confval:`extensions` configuration
+value in `docs/conf.py`_.
+
+.. _external-references:
+
+References to other packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intersphinx_ allows the automatic generation of links to the
+documentation of objects in other projects. This cross-package linking
+is made possible with the `sphinx.ext.intersphinx` extension.
+
+When we include ``astropy.units.Quantity`` in the documentation, it will
+show up as `astropy.units.Quantity` with a link to the appropriate page
+in Astropy documentation. Similarly, ``~astropy.units.Quantity`` will
+show up as `~astropy.units.Quantity`.
+
+The mappings are defined in the :confval:`intersphinx_mapping`
+dictionary in `docs/conf.py`_, and include Python_, NumPy_, SciPy_,
+Astropy_, pandas_, Sphinx_, and `sphinx_automodapi`_.
+
+New source packages may be added, but please verify that references to a
+function or class in that package show up correctly in `PlasmaPy's
+documentation`_. The name of the package does not always link as
+expected.
+
+.. hint::
+
+   If a cross-link is not working as expected this is usually due to one
+   of the following reasons:
+
+   * A typo;
+   * The package not being defined in :confval:`intersphinx_mapping`, or
+   * The referenced source package not properly or fully indexing their
+     own code, which is common in Python_ packages.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST_ allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
```

Comment:
```suggestion
.. code-block:: rst

   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`

Here whenever ``|Particle|`` is used Sphinx_ will replace it with 
`` `~plasmapy.particles.particle_class.Particle` `` during build time.
```

Nowhere in this section did you explicitly say Sphinx substitutes X for Y during building.  I wanted to directly say that.